homeassistant/components/edl21/config_flow.py

```diff
@@ -0,0 +1,52 @@
+"""Config flow for EDL21 integration."""
+from typing import Any
+
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_NAME
+from homeassistant.data_entry_flow import FlowResult
+
+from .const import CONF_SERIAL_PORT, DEFAULT_NAME, DOMAIN
+
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_NAME, default=DEFAULT_NAME): str,
+        vol.Required(CONF_SERIAL_PORT): str,
+    }
+)
+
+
+class EDL21ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """EDL21 config flow."""
+
+    VERSION = 1
+
+    async def async_step_import(self, import_config: dict[str, Any]) -> FlowResult:
+        """Import a config entry from configuration.yaml."""
+
+        self._async_abort_entries_match(
+            {CONF_SERIAL_PORT: import_config[CONF_SERIAL_PORT]}
+        )
+        if import_config[CONF_NAME]:
+            self._async_abort_entries_match({CONF_NAME: import_config[CONF_NAME]})
```

Comment:
As I have said before, I really don't think this makes sense here.
The entities have a `unique_id`, so they can be renamed in the UI.
I would maybe even completely drop `CONF_NAME` from the config.

Reply:
If you have two smart meter of the same type, both smart meter will 'create' the same set of entities (ten each in my case). When the entities have the same name (pair wise) there is no way of telling which entity belongs to which smart meter. The `unique_id` does not help to find out which entity belongs to which smart meter. I think this is the reason why the `CONF_NAME` was introduced in the first place.

Reply:
I think that:
- it is OK to ask for the name, but I think it should be stored only in the title of the config entry (not in the data)
- when the entities are created, it can use the title of the config entry for the initial name of the entities

In the future, when you add `DeviceInfo` it will become even clearer...

Reply:
Ok, I think I (finally) understood...
- For the 'legacy' configuration, the name will be added to the entity names as before.
- For new installations the ``CONF_NAME`` property will be removed. When having multiple smart meter, I have to (manually) rename the entities.
- After the introduction of the DeviceInfo (upcoming PR) the renaming can be done by just renaming the Device which will also rename all entities for that device.

Reply:
👍 adding DeviceInfo is usually a small change. 
If you get both merged before last Wednesday of the month there will probably be no user facing the second option (except nightly users).

Reply:
Since we are now in agreement, can you remove these two lines?
```suggestion
```